network/patton/smartnode/snmp/mode/system.pm

```diff
@@ -0,0 +1,261 @@
+};
+
+my $memoryPoolEntry_oid = '.1.3.6.1.4.1.1768.100.70.20.2.1';
+my $mappingMemory = {
+    display => { oid => '.1.3.6.1.4.1.1768.100.70.20.2.1.1' }, # memDescr
+    memory_total_bytes => { oid => '.1.3.6.1.4.1.1768.100.70.20.2.1.2' }, # memTotalBytes
+    memory_usage_bytes => { oid => '.1.3.6.1.4.1.1768.100.70.20.2.1.3' }, # memAllocatedBytes
+    memory_free_bytes => { oid => '.1.3.6.1.4.1.1768.100.70.20.2.1.4' } # memFreeBytes
+};
+
+my $tempProbeEntry_oid = '.1.3.6.1.4.1.1768.100.70.30.2.1';
+my $mappingTemperature = {
+    display => { oid => '.1.3.6.1.4.1.1768.100.70.30.2.1.1' }, # tempProbeDescr
+    probe_temperature_celsius => { oid => '.1.3.6.1.4.1.1768.100.70.30.2.1.2' } # currentDegreesCelsius
+};
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $snmp_results = $options{snmp}->get_multiple_table(oids => [
+        { oid => $cpuEntry_oid },
+        { oid => $memoryPoolEntry_oid },
+        { oid => $tempProbeEntry_oid }
+    ], nothing_quit => 1);
+
+    my $numCpu = 1;
+    my $cpuTotal = 0;
+    my $cpuEntry = $options{snmp}->map_instance(mapping => $mappingCpu, results => $snmp_results->{$cpuEntry_oid}, instance => $numCpu);
+    while (defined($cpuEntry->{display})) {
+        $cpuTotal += $cpuEntry->{cpu_utilization_current_percentage};
+        $self->{cpu}->{ $cpuEntry->{display} } = $cpuEntry;
+        $numCpu++;
+        $cpuEntry = $options{snmp}->map_instance(mapping => $mappingCpu, results => $snmp_results->{$cpuEntry_oid}, instance => $numCpu);
+    }
+
+    if ($numCpu > 1) {
+        $self->{cpu_average} = {
+            average => $cpuTotal / ($numCpu - 1),
+            count => ($numCpu - 1)
+        };
+    }
+
+    my $numMem = 1;
+    my $memoryPoolEntry = $options{snmp}->map_instance(mapping => $mappingMemory, results => $snmp_results->{$memoryPoolEntry_oid}, instance => $numMem);
+    while (defined($memoryPoolEntry->{display})) {
+        if ($memoryPoolEntry->{memory_total_bytes} > 0) {
+            $memoryPoolEntry->{memory_usage_percentage} = $memoryPoolEntry->{memory_usage_bytes} * 100 / $memoryPoolEntry->{memory_total_bytes};
+            $memoryPoolEntry->{memory_free_percentage} = $memoryPoolEntry->{memory_free_bytes} * 100 / $memoryPoolEntry->{memory_total_bytes};
+        } else {
+            $memoryPoolEntry->{memory_usage_percentage} = 0;
+            $memoryPoolEntry->{memory_free_percentage} = 0;
+        }
+
+        $self->{memory}->{ $memoryPoolEntry->{display} } = $memoryPoolEntry;
+        $numMem++;
+        $memoryPoolEntry = $options{snmp}->map_instance(mapping => $mappingMemory, results => $snmp_results->{$memoryPoolEntry_oid}, instance => $numMem);
+    }
+
+    my $numTemp = 1;
+    my $tempProbeEntry = $options{snmp}->map_instance(mapping => $mappingTemperature, results => $snmp_results->{$tempProbeEntry_oid}, instance => $numTemp);
+    while (defined($tempProbeEntry->{display})) {
```

Comment:
the instance can start with another value than 1. It's not the right way to do the parsing.